src/pages/TCY/components/Claim/ClaimSelect.tsx

```diff
@@ -103,21 +104,31 @@ export const ClaimSelect: React.FC<TCYRouteProps & { activeAccountNumber: number
   headerComponent,
   activeAccountNumber,
 }) => {
-  const { isOpen, onOpen, onClose } = useDisclosure()
-  const [activeClaim, setActiveClaim] = useState<Claim | undefined>()
+  const navigate = useNavigate()
+  const location = useLocation()
+  const claimsQueries = useTCYClaims(activeAccountNumber)
+  const claims = claimsQueries
+    .map(query => query.data)
+    .flat()
+    .filter(isSome)
+
+  const maybeClaimMatch = matchPath('/tcy/claim/:l1_address', location.pathname)
+  const l1_address = maybeClaimMatch?.params?.l1_address
+  // Prefer claim from navigation state, fallback to lookup
```

Comment:
Precisely what this says - passed with `navigate()` as state, but also works using the URL pattern matching